### DISCUSSION TITLE
Wow this is a nice framework!

### DISCUSSION THREAD

**Discussion by lawso017** [odoocker/odoocker#14] [Q&A]
Hi and wow -- congrats on putting together such a well-thought-out framework. Really impressed with your tech stack choices, was working along the exact same lines using KeyDB instead of Redis and updating CampToCamp's S3 storage addon for v16.  Integrating minio is an especially nice touch!

Working on deploying Odoo into a Kubernetes cluster with Percona's Postgresql operator managing the database cluster. Starting with your framework and will let you know how it goes...

**Comment by yhaelopez** [comment]
I'm so glad and excited you are building something on top of this!

Let me know if I can help you in any way. Any questions will be answered. Maybe we can catch up in a meeting some day if you are willing to. So we both can share ideas and resolve questions!

We are planning to replace nginx proxy and acme companion with nginx proxy manager, and also add the choice to use traefik instead, to have a better and selectable stack.

Hope I hear from your project soon!

**Comment by lawso017** [comment]
Ok after a day of exploring, this is indeed a robust framework for managing development & testing environments. The primary limitation I'm seeing when looking at staging / production deployments is that we cannot store any secrets in our Docker images....  best practice is that we should pass secrets into the environment using the appropriate secret management system (ie Kubernetes Secrets for example).

That means that our entrypoint.sh file should allow us to override the database connection information for example. Also, things like the ADMIN_PASSWD are probably best not stored in the image's odoo.conf file. Let me know if that makes sense, and if you have any thoughts...

**Reply by yhaelopez** [reply]
Yes, this makes sense to me. You can propose an initial solution we can work it and polish any details to integrate it to the project, keeping the simplicity of development stages, but being able to do this with secret manager.
